pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -324,6 +324,25 @@ def __warn_deprecated_monitor_key(self):
                 """
             rank_zero_warn(m)
 
+    def _validate_condition_metric(self, logs):
+        monitor_val = logs.get(self.monitor)
+
+        if monitor_val is None:  # self.monitor wasn't found in the metrics dict
+            error_msg = (f'Checkpointing conditioned on metric `{self.monitor}`'
+                         f' which is not available. Either add `{self.monitor}` to the return of '
+                         f' validation_epoch end or modify your CheckPoint callback to use any of the '
+                         f'following: `{"`, `".join(list(logs.keys()))}`')
+
+            if self.verbose > 0:
+                rank_zero_warn(error_msg, RuntimeWarning)
+
+            return False
+
+        return True
+
+    def _run_checkpoint_check(self, metrics):
+        self._validate_condition_metric(metrics)
```

Comment:
I think `_validate_condition_metric` is enough here, why do we need a function to call another function what is actually doing something?